packages/ckeditor5-editor-classic/tests/classiceditorui.js

```diff
@@ -756,31 +756,141 @@ describe( 'Focus handling and navigation between editing root and editor toolbar
 
 			setModelData( editor.model, '<paragraph>foo[]</paragraph>' );
 
-			pressEsc();
+			pressEsc( editor );
 
 			sinon.assert.notCalled( domRootFocusSpy );
 			sinon.assert.notCalled( toolbarFocusSpy );
 		} );
 	} );
+} );
+
+describe( 'Focus handling and navigation between editing root and editor menu bar', () => {
+	let editorElement, editor, ui, menuBarView, domRoot;
+
+	testUtils.createSinonSandbox();
 
-	function pressAltF10() {
-		editor.keystrokes.press( {
-			keyCode: keyCodes.f10,
-			altKey: true,
-			preventDefault: sinon.spy(),
-			stopPropagation: sinon.spy()
+	beforeEach( async () => {
+		editorElement = document.body.appendChild( document.createElement( 'div' ) );
+
+		editor = await ClassicEditor.create( editorElement, {
+			plugins: [ Paragraph, Image, ImageToolbar, ImageCaption ],
+			toolbar: [ 'imageTextAlternative' ],
+			image: {
+				toolbar: [ 'toggleImageCaption' ]
+			}
 		} );
-	}
 
-	function pressEsc() {
-		editor.keystrokes.press( {
-			keyCode: keyCodes.esc,
-			preventDefault: sinon.spy(),
-			stopPropagation: sinon.spy()
+		domRoot = editor.editing.view.domRoots.get( 'main' );
+
+		ui = editor.ui;
+		menuBarView = ui.view.menuBarView;
+	} );
+
+	afterEach( () => {
+		editorElement.remove();
+
+		return editor.destroy();
+	} );
+
+	describe( 'Focusing menu bar on Alt+F9 key press', () => {
+		beforeEach( () => {
+			ui.focusTracker.isFocused = true;
+			ui.focusTracker.focusedElement = domRoot;
 		} );
-	}
+
+		it( 'should focus the menu bar when the focus is in the editing root', () => {
+			const spy = testUtils.sinon.spy( menuBarView, 'focus' );
+
+			setModelData( editor.model, '<paragraph>foo[]</paragraph>' );
+
+			ui.focusTracker.isFocused = true;
+			ui.focusTracker.focusedElement = domRoot;
+
+			// Focus the menu bar.
+			pressAltF9( editor );
+
+			sinon.assert.calledOnce( spy );
+		} );
+
+		it( 'should do nothing if the menu bar is already focused', () => {
+			const domRootFocusSpy = testUtils.sinon.spy( domRoot, 'focus' );
+			const menuBarFocusSpy = testUtils.sinon.spy( menuBarView, 'focus' );
+
+			setModelData( editor.model, '<paragraph>foo[]</paragraph>' );
+
+			// Focus the menu bar.
+			pressAltF9( editor );
+			ui.focusTracker.focusedElement = menuBarView.element;
+
+			// Try Alt+F9 again.
+			pressAltF9( editor );
+
+			sinon.assert.calledOnce( menuBarFocusSpy );
+			sinon.assert.notCalled( domRootFocusSpy );
+		} );
+	} );
+
+	describe( 'Restoring focus on Esc key press', () => {
```

Comment:
**This is a follow-up**: We're going to need this kind of logic for the menu bar as well. 

https://github.com/ckeditor/ckeditor5/blob/76919d55c7825bd4401d4a473d6415922f80b49e/packages/ckeditor5-ui/src/editorui/editorui.ts#L437-L464

I coded the prototype as fast as possible so I didn't consider cases where the focus must get back to some other place (e.g. source editing). 

I suppose `EditorUI#addMenuBar` corresponding to `addToolbar` would be a good idea. Because currently, the keystroke handling code for the menu bar is in `ClassicEditorUI`, and if sb wanted to build their editor, they wouldn't be able to add a menu bar without duplicating the code.

cc @DawidKossowski